lib/internal/streams/legacy.js

```diff
@@ -111,4 +112,24 @@ function prependListener(emitter, event, fn) {
     emitter._events[event] = [fn, emitter._events[event]];
 }
 
+let defaultHighwaterMark = null;
+ObjectDefineProperty(Stream, 'defaultHighwaterMark', {
+  get () {
+    return defaultHighwaterMark || Buffer.poolSize || 16 * 1024;
+  },
+  set (value) {
+    defaultHighwaterMark = value;
+  }
+});
+
+let defaultObjectModeHighwaterMark = null;
+ObjectDefineProperty(Stream, 'defaultObjectModeHighwaterMark', {
+  get () {
+    return defaultObjectModeHighwaterMark || 16;
```

Comment:
0 defaulting to 16 and -1 not doing anything meaningful is undocumented, I think.
Of those, `0` actually does have some sense, and -1 doesn't.

Perhaps, set the default value on the `let` line and then check for valid values in `set()` (see above)?